src/openrct2/peep/Guest.cpp

```diff
@@ -2503,6 +2510,10 @@ static Vehicle* peep_choose_car_from_ride(Peep* peep, Ride* ride, std::vector<ui
  */
 static void peep_choose_seat_from_car(Peep* peep, Ride* ride, Vehicle* vehicle)
 {
+    if (vehicle == nullptr)
```

Comment:
It's kind of unrelated to this PR, but maybe we should check for the other pointers